### PR TITLE
Force apps missing namespaces to use the default namespace

### PR DESCRIPTION
**Pull request by babarinde**
When a namesapce is defined by using `dokku scheduler-kubernetes:set $APP namespace test`, subsequent request via kubectl calls will use this set namespace for app like `kubectl --namespace=test apply -f ...`
However, when a namespace is not defined for an app subsequent kubectl calls will use  for example `kubectl apply -f ...` without specifying a namespace.
The assumption here is that  kubectl will use the `default` namespace, however if one has changed context  via `kubectl config set-context --current --namespace=pxc`, all such kubectl calls missing the --namespace flag will use the pxc namespace instead of default.
The purpose of this PR is to force all such kubectl calls for apps without a namespace defined to use the `default` namespace by default

### CODE REVIEW

**Comment by josegonzalez** [comment]
Can someone delete the `default` namespace?

**Comment by babarinde** [comment]
According to [this](https://cloud.google.com/blog/products/gcp/kubernetes-best-practices-organizing-with-namespaces)

> There is absolutely nothing special about this Namespace, except that the Kubernetes tooling is set up out of the box to use this namespace and you can’t delete it. 

 you can't delete the default namespace

File: internal-functions [hunk 1]
```diff
@@ -32,6 +32,8 @@ cmd-scheduler-kubernetes-rolling-update() {
   if [[ -n "$NAMESPACE" ]]; then
     KUBE_ARGS+=("--namespace=$NAMESPACE")
     fn-scheduler-kubernetes-ensure-namespace "$NAMESPACE" >/dev/null
+  else
+    KUBE_ARGS+=("--namespace=default")
```
Comment (josegonzalez) [inline internal-functions]: Maybe change this:

```
NAMESPACE="$(fn-plugin-property-get "scheduler-kubernetes" "$APP" "namespace" "")"
```

to this:

```
NAMESPACE="$(fn-plugin-property-get "scheduler-kubernetes" "$APP" "namespace" "default")"
```

and have `fn-scheduler-kubernetes-ensure-namespace` return immediately if the `NAMESPACE` argument is `default`? That would minimize the code here.

Comment (babarinde) [reply internal-functions]: If i understood you correctly, the above becomes something like this
```
NAMESPACE="$(fn-plugin-property-get "scheduler-kubernetes" "$APP" "namespace" "default")"
KUBE_ARGS+=("--namespace=$NAMESPACE")
fn-scheduler-kubernetes-ensure-namespace "$NAMESPACE" >/dev/null
```

and in `fn-scheduler-kubernetes-ensure-namespace` we have
```
...
if [[  "$NAMESPACE" = "default" ]]; then
    return
fi
SIGIL_PARAMS=(NAME="$NAMESPACE")
...
```

Comment (josegonzalez) [reply internal-functions]: yeah more or less :)

Comment (babarinde) [reply internal-functions]: changes effected

**Pull request merged**